src/utils/url.ts

```diff
@@ -63,8 +49,18 @@ export const openUrl = (url: string) => {
 };
 
 export const isValidUrl = (url: string): boolean => {
+  const checkUrl = (url: string): boolean =>
+    VALID_URL_PROTOCOLS.includes(new URL(url).protocol);
+  const relativeLinksRegex = /^(\/in\/|\/v\/).*/;
+  const shortUrlRegex = /^\/\/.*/;
   try {
-    return VALID_URL_PROTOCOLS.includes(new URL(url).protocol);
+    if (new RegExp(relativeLinksRegex).test(url)) {
+      return checkUrl(window.location.origin + url);
+    }
+    if (new RegExp(shortUrlRegex).test(url)) {
+      return checkUrl(window.location.protocol + url);
+    }
+    return checkUrl(url);
   } catch (e) {
```

Comment:
The changes here look quite convoluted for what I believe should be a relatively simple fix.

---

Regex explanations (where `*` in the below descriptions denotes anything):

- `relativeLinksRegex`: matches `/in/*` and `/v/*` urls
- `shortUrlRegex`: matches `//*` urls

---

Looking at the docs for `URL`, it seems that by including the `base` param, it can handle relative urls:

- https://developer.mozilla.org/en-US/docs/Web/API/URL/URL#parameters
  - If `url` is a relative URL, `base` is required, and will be used as the base URL. If `url` is an absolute URL, a given `base` will be ignored.
  - https://developer.mozilla.org/en-US/docs/Web/API/URL/URL#examples

Playing around with that:

```javascript
console.log(window.location.origin)
// https://developer.mozilla.org

new URL('//', window.location.origin).toString()
// Uncaught TypeError: Failed to construct 'URL': Invalid URL

new URL('//foo', window.location.origin).toString()
// "https://foo/"

new URL('//foo.com', window.location.origin).toString()
// "https://foo.com/"

new URL('/in/aaa', window.location.origin).toString()
// "https://developer.mozilla.org/in/aaa"

new URL('/v/bbb', window.location.origin).toString()
// "https://developer.mozilla.org/v/bbb"
```
  
So I believe that removing all of the new code here, and just changing line 67 to something like the following might be sufficient:

```typescript
return VALID_URL_PROTOCOLS.includes(new URL(url, window.location.origin).protocol);
```

Note: I haven't checked through all of the reported errors to see what other URL patterns exist, so it may need to be slightly more complex than this

Reply:
you are right. It's the most suitable way to validate inputting URLs. Have checked the error cases from the Bugsnag

Reply:
> Have checked the error cases from the Bugsnag

When you say that, do you mean that you've checked against the error cases and this is sufficient to handle them? Or do you still need to do something else to improve this check further?

Reply:
no, it's fine, have already checked the error cases

Reply:
Awesome :)